atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -580,6 +581,11 @@ private RaftRole createRole(final RaftServer.Role role) {
     }
   }
 
+  private ElectionTimer createElectionTimer(final Runnable triggerElection, final Logger log) {
```

Comment:
Off topic: Do you know why we in atomix often have these logger parameters? In other code places we normally have the Loggers class which we use to get the right logger :thinking:

Reply:
Here I added it as a parameter because it is nice to use the same logger as the follower role because it has both the partition id and the role added to the logger context. So it is easier to filter the logs when debugging. We can built it here also, but then the logic will be duplicated.
I think this might be the reason why we have this pattern in other places. I don't remember when we started doing it.